### PR TITLE
Theorems for Godel-sets

### PR DESCRIPTION
**Pull request by avekens**
As indicated in issue #87 by @tirix , there is material for starting with the proof of Gödel's incompleteness theorem in @digama0's mathbox. I had a closer look at it, and found it quite interesting. Unfortunately, there were only definitions (very complex, using recursion, e.g., df- sat), and no (basic) theorems. Therefore I added some basic theorems to verify the definitions and to make them more understandable. Since the definitions are in  @digama0's mathbox, I had to add the theorems also in this mathbox. @digama0 I hope this is OK for you. 

Some proofs are brute force attempts (I wanted to have the results as soon as possible), so there is room to make them shorter/more elegant.

I cannot promise if and how far I will dive into the proof of Gödel's incompleteness theorem, but I think I will continue to analyse the other definitions in this context more deeply. @digama0 is there additional material for these definitions available? Bibliographic references? Unpublished theorems you proved already?

Details:
main:
* ~sels (formerly ~bj-sels) moved from BJ's mathbox
* new theorems ~opabex3rd, ~omsucelsucb added

MC's mathbox:
* section "Godel-sets of formulas" split into two parts
* comment for ~df-fmla adjusted/enhanced
* basic theorems for Godel-set of membership and Godel-set of universal quantification added
*  basic theorems for satisfaction predicate (Sat) added
* basic theorems for Goedel formulas (Fmla) added

### CODE REVIEW

**Comment by avekens** [comment]
I made a typo in the date contained in "(Contributed by ...)", which was not detected by metamath.exe, verify markup.  metamath-knife, however, detected it, but gave a misleading error message:

```
/target/release/metamath-knife --verify \
warning: No contribution comment
      --> ./set.mm:505703:5
       |
505703 |     satf0 $p |- ( (/) Sat (/) ) = ( rec ( ( f e. _V |-> ( f u.
       |     ----- No (Contributed by...) provided for this statement
```

There was a (Contributed by...), but it was erroneous...

**Comment by tirix** [comment]
> I made a typo in the date contained in "(Contributed by ...)", which was not detected by metamath.exe, verify markup. metamath-knife, however, detected it, but gave a misleading error message.

Good catch! We should probably change the error message to "Missing or malformed  contribution comment".
Can you open an issue in `metamath-knife` for that?

**Comment by avekens** [comment]
> > I made a typo in the date contained in "(Contributed by ...)", which was not detected by metamath.exe, verify markup. metamath-knife, however, detected it, but gave a misleading error message.
> 
> Good catch! We should probably change the error message to "Missing or malformed contribution comment". Can you open an issue in `metamath-knife` for that?

Issue https://github.com/metamath/metamath-knife/issues/125 is opened.

**Review by tirix — APPROVED** [review]
Nice!

There is something called "[Gödel Constuctible Sets](https://encyclopediaofmath.org/wiki/G%C3%B6del_constructive_set)", I wonder if this is what @digama0 was formalizing, rather than Gödel Numbers.

If so, this might or might not be useful for the Incompleteness Theorem.

File: set.mm [hunk 1]
```diff
@@ -505582,15 +505610,553 @@ valuation of the variables (v_0 ` = ( S `` (/) ) ` , v_1 ` = ( S `` 1o
...
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $( Introduce new constant symbols. $)
+  $c =g $. $( Godel-set of equality $)
+  $c /\g $. $( Godel-set of conjunction $)
+  $c -.g $. $( Godel-set of negation $)
+  $c ->g $. $( Godel-set of implication $)
+  $c <->g $. $( Godel-set of equivalence $)
+  $c \/g $. $( Godel-set of disjunction $)
```
[the remark this answers is not in the document]
Comment (digama0) [reply set.mm:506157]: These are functions defined in terms of the primitives, to make it easier to express formulas using these definitions.

**Comment by digama0** [comment]
I forget what precisely I was formalizing, I think it was model theory generally, and/or its connection to metamath formal systems in the following section. Because the formulation of formulas and satisfaction is tailored to ZF - it is sufficient to state things about ZF provability or unprovability but might be a bit too specialized for general FOL models.

File: set.mm [hunk 2]
```diff
@@ -71610,6 +71619,28 @@ as a separate axiom in an axiom system (such as Kunen's) that uses this
       CEVQWLNLFAYDCEAWFMWINLWJNLYDWEVSGWIWJNNVTVRWACEWKNNWBWCWD $.
   $}
 
+  ${
+    $d A x y v w z $.  $d ps v w z $.  $d ph y $.
+    opabex3rd.1 $e |- ( ph -> A e. _V ) $.
+    opabex3rd.2 $e |- ( ( ph /\ y e. A ) -> { x | ps } e. _V ) $.
+    $( Existence of an ordered pair abstraction if the second components are
+       elements of a set.  (Contributed by AV, 17-Sep-2023.) $)
+    opabex3rd $p |- ( ph -> { <. x , y >. | ( y e. A /\ ps ) } e. _V ) $=
```
[the remark this answers is not in the document]
Comment (avekens) [reply set.mm:71628]: I think the official name is "ordered-pair class abstraction" (see section "Ordered-pair class abstractions (class builders)"), although "class abstraction of (ordered) pairs" and "ordered pair abstraction" are also often used. The harmonization of these names should be performed in a separate PR.

File: set.mm (hunk 1 above)
[the remark this answers is not in the document]
Comment (tirix) [reply set.mm:506157]: Got it, thank you!

**Comment by avekens** [comment]
@digama0 please approve this PR if it is OK for you that I put my theorems in your mathbox.

**Review by digama0 — APPROVED** [review]

**Pull request merged**